### PR TITLE
Pipeline V1

### PR DESCRIPTION
**Pull request by pjcggf**
# Pipeline V1

### Data split 
`data_split(data, list_y, list_X, classification)`

Permet de récupérer un dataset (**au format pandas DataFrame**) et récupérer les 2 couples X et y train et test splitté.
<ins>Paramètres obligatoires:</ins> 
-  `data`: dataset (`pd.DataFrame`)
- `list_y`: liste des colonnes à conserver pour y (`list`)
- `list_X`: liste des colonnes à <ins>**DROP**</ins> pour X (`list`)
- `classification`: type de tâche (classification=`True/False`)

### Pipeline 
`pipeline(scaler='StandardScaler',model_path='sklearn.linear_model',model_name='LinearRegression',params={},classification=False)`

Permet de récupérer un pipeline non fitté.
<ins>Paramètres optionnels:</ins>
- `scaler`: nom du scaler à utiliser dans le pipe (`str`, ex: "RobustScaler")
- `model_path`: chemin d'accès à la famille du modèle à utiliser dans le pipe (`str`, ex: "sklearn.neighbors")
- `model_name`: nom du modèle à utiliser (`str`, ex:"KNeighborsRegressor")
- `params`: dictionnaire contenant les hyperparamètres du modèle (`dict`, ex:`{"n_neighbors": 5, "weights": 'uniform'}`)
- `classification`: type de tâche (classification=`True/False`)

Si **`classification=True`** le modèle est inséré dans un `MultiOutputRegressor`

### Cross validation score

`cross_validation_score(pipe, data, list_y, list_X, cv=5, classification=False)`

Permet de récupérer une liste des scores et les X et y de train et de test.
<ins>Paramètres obligatoires:</ins> 
- `pipe`: pipeline utilisé pour la cross validation(`sklearn.pipeline.Pipeline`)
- `data`: dataframe pour la cross validation(`pd.DataFrame`)
- `list_y`: liste des colonnes à conserver pour y (`list`)
- `list_X`: liste des colonnes à <ins>**DROP**</ins> pour X (`list`)
<ins>Paramètres optionnels:</ins>
- `cv`: nombre de fold demandé pour la cross validation(`int`)
- `classification`: type de tâche (classification=`True/False`)

### Get pipeline

`get_train_pipeline(data, list_y, list_X, cv=5,
                        classification=False,
                        scaler='StandardScaler',
                        model_path='sklearn.linear_model', model_name='LinearRegression', params={},  return_full_set=False):
    pipe = pipeline(scaler, model_path, model_name, params, classification)`

Permet de récupérer un pipeline fitté, la liste des scores de la cross validation du modèle, un X_test/y_test et X_train/y_train si `return_full_set=True` (correspondant aux données du fit pour éviter le data leakage)
<ins>Paramètres obligatoires:</ins> 
- `data`: dataframe pour la cross validation(`pd.DataFrame`)
- `list_y`: liste des colonnes à conserver pour y (`list`)
- `list_X`: liste des colonnes à <ins>**DROP**</ins> pour X (`list`)
<ins>Paramètres optionnels:</ins>
- `cv`: nombre de fold demandé pour la cross validation(`int`)
- `classification`: type de tâche (classification=`True/False`)
- `scaler`: nom du scaler à utiliser dans le pipe (`str`, ex: "RobustScaler")
- `model_path`: chemin d'accès à la famille du modèle à utiliser dans le pipe (`str`, ex: "sklearn.neighbors")
- `model_name`: nom du modèle à utiliser (`str`, ex:"KNeighborsRegressor")
- `params`: dictionnaire contenant les hyperparamètres du modèle (`dict`, ex:`{"n_neighbors": 5, "weights": 'uniform'}`)
- `return_full_set`: indique si l'on souhaite récupérer uniquement le X_test/y_test (`False`) ou X_train/y_train et X_test/y_test(`True`)(`bool`)

## Prochain développement

Ajout d'un custom scorer pour cross valider sur l'accuracy les modèles regression ou classification

### CODE REVIEW

**Review by AxelAvocat-Benan — APPROVED** [review]

**Comment by pjcggf** [comment]
# Pipeline V2

## Grid Search

`grid_search(pipe, X_train,  y_train, cv=5)`

Permet de sélectionner les paramètres du pipe que l'on souhaite grid search puis récupérer un modèle avec les paramètres ayant obtenu le meilleur score.

<ins>Paramètres obligatoires:</ins> 
-  `pipe`: pipe à grid search (`pd.DataFrame`)
- `X_train`: X pour entrainement (`pd.DataFrame`)
- `y_train`: y pour entrainement (`pd.DataFrame`)

<ins>Paramètres optionnel:</ins> 
- `cv`: nombre de fold demandé pour la cross validation(`int`)

**Comment by pjcggf** [comment]
## Ajout Docstring

Docstring ajouté dans chaque fonction pour faciliter la compréhension des différents arguments de chacune

**Review by AxelAvocat-Benan — APPROVED** [review]

**Review by mpnicolas — APPROVED** [review]
vu

**Review by NicolasJalbau — APPROVED** [review]
